Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/ThemeHelper.java

```diff
@@ -0,0 +1,125 @@
+        switch (currentNightMode)
+        {
+          case Configuration.UI_MODE_NIGHT_NO:
+            activity.getWindow().setStatusBarColor(
+                    activity.getResources().getColor(R.color.m3_sys_color_dynamic_light_surface));
+            break;
+          case Configuration.UI_MODE_NIGHT_YES:
+            activity.getWindow().setStatusBarColor(
+                    activity.getResources().getColor(R.color.m3_sys_color_dynamic_dark_surface));
+            break;
+        }
+        break;
+
+      case MATERIAL_DEFAULT:
+        activity.setTheme(R.style.Theme_Dolphin_Main_Material);
+        activity.getWindow()
+                .setStatusBarColor(activity.getResources().getColor(R.color.dolphin_surface));
+        break;
+
+      case GREEN:
+        activity.setTheme(R.style.Theme_Dolphin_Main_Green);
+        activity.getWindow()
+                .setStatusBarColor(activity.getResources().getColor(R.color.green_surface));
+        break;
+
+      case PINK:
+        activity.setTheme(R.style.Theme_Dolphin_Main_Pink);
+        activity.getWindow()
+                .setStatusBarColor(activity.getResources().getColor(R.color.pink_surface));
+        break;
+    }
+
+    // Since the top app bar matches the color of the status bar, devices below API 23 have to get a
+    // black status bar since their icons do not adapt based on background color
+    if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M)
+    {
+      activity.getWindow()
+              .setStatusBarColor(activity.getResources().getColor(android.R.color.black));
+    }
+  }
+
+  public static void saveTheme(Activity activity, int themeValue)
+  {
+    mSharedPreferences = activity.getSharedPreferences(THEME, Context.MODE_PRIVATE);
+    mSharedPreferences.edit().putInt(CURRENT_THEME, themeValue).apply();
+  }
+
+  public static void deleteThemeKey(Activity activity)
+  {
+    mSharedPreferences = activity.getSharedPreferences(THEME, Context.MODE_PRIVATE);
+    mSharedPreferences.edit().remove(CURRENT_THEME).apply();
+    activity.setTheme(R.style.Theme_Dolphin_Main);
+    activity.recreate();
+  }
+
+  public static void setCorrectTheme(MainActivity activity)
+  {
+    int currentTheme = activity.getThemeId();
+    setTheme(activity);
+
+    if (currentTheme != activity.getThemeId())
+    {
+      activity.recreate();
+    }
+  }
+
+  public static void checkThemeSetting(Activity activity)
+  {
+    mSharedPreferences = activity.getSharedPreferences(THEME, Context.MODE_PRIVATE);
+    if (IntSetting.MAIN_INTERFACE_THEME.getIntGlobal() !=
+            mSharedPreferences.getInt(CURRENT_THEME, DEFAULT))
+    {
+      mSharedPreferences.edit()
+              .putInt(CURRENT_THEME, IntSetting.MAIN_INTERFACE_THEME.getIntGlobal())
+              .apply();
+      activity.recreate();
+    }
+  }
+
+  public static int dpToPx(float dp, Resources resources)
+  {
+    return (int) (dp * resources.getDisplayMetrics().density + 0.5f);
+  }
```

Comment:
This doesn't really have anything to do with themes. If you really need it (would using layout XML files let us avoid having this function?), it probably fits better to put in a different file.